### ISSUE TITLE
Preserve white space in Flax views

### ISSUE THREAD

**Issue by essenciary** [GenieFramework/Genie.jl#184]
**Describe the bug**
In the resources/views/, it appears like the parser removes spaces, and even explicit spaces with &nbsbare ignored. For example, <strong>Test &nbsp Test &nbsp </strong>&nbsp Test Testresults in the following. Any way to force a space? I tried the _ prefix as for reserved word like for but without luck.

https://files.gitter.im/essenciary/Genie.jl/KBFR/image.png

**Comment by jeremiedb** [comment]
Not sure if related, but I also noticed a strange behavior in the case of generated radio buttons. 
If the label/input aren't nested into a `<div>`, only the radio button would appear. 
And if the input is put first, then  it's only the label that shows. Like if only the last line is parsed. Maybe is there's the assumption in the parsing about the `<input>` being expected to be closed with a `</input>` ? 

```html
<% @foreach(@vars(:aggs)) do agg %>
<% if agg == @vars(:agg) %>
	<div class="form-group">
		<label class="form-check-label" _for=$(agg)>$(agg): </label>
		<input class="form-check-input" type="radio" name="agg" id=$(agg) value=$(agg) checked>
	</div>
<% else %>
<div class="form-group">
	<label class="form-check-label" _for=$(agg)>$(agg): </label>
	<input class="form-check-input" type="radio" name="agg" id=$(agg) value=$(agg)>
</div>
<% end %>
<% end %>
```

**Comment by essenciary** [comment]
@jeremiedb the whitespace issue is now fixed and will be released in v0.20 (probably v1.0-rc). It's now residing on the #no_nullables branch. 

Re the 2nd issue, this is known but I don't have a good solution yet. Keep in mind that the HTML view code in converted into Julia expressions. So you end up with something like: 

```julia
if agg == @vars(:agg) 
  Html.input(...)
  Html.label(...)
end
```

The `if` statement automatically returns the result of the last expression, so you only end up with the output of the last expression _only_. Probably the way to go will be to introduce a `@if` macro. For now, you can wrap the code into an array, ie: 

```html
<% if agg == @vars(:agg); [ %>
	<div class="form-group">
		<label class="form-check-label" _for=$(agg)>$(agg): </label>
		<input class="form-check-input" type="radio" name="agg" id=$(agg) value=$(agg) checked>
	</div>
<% ]end %>
```

This will result in returning the array with all its elements from the `if` expression and works as expected.

**Comment by jeremiedb** [comment]
Thanks for the feedback! 
If I want to better understand how the parsing of views is performed, should I look into src/Flax.jl?

**Comment by essenciary** [comment]
You're welcome!
Yes, Renderer -> Flax -> HTMLRenderer
Definitely go straight to the #no_nullables branch (or wait to be merged into #master in a couple of days hopefully). This also comes with new tests for the rendering engine so you can check that too.

**Comment by essenciary** [comment]
This is now out on v0.20 (still no fix for if but I'll add it for v0.21 maybe)

**Comment by essenciary** [comment]
I'm going to close this as it's correctly solved. We'll have to address the `if` blocks rendering somewhere else (if a satisfactory solution is found).
